server/geoip_service.py

```diff
@@ -189,18 +195,24 @@ def load_db(self) -> None:
         """
         Loads the database into memory.
         """
+        # Set the time first, if the file is corrupted we don't need to try
+        # loading it again anyways
+        self.db_update_time = datetime.now()
+
         try:
-            # Set the time first, if the file is corrupted we don't need to try
-            # loading it again anyways
-            self.db_update_time = datetime.now()
-            self.db = geoip2.database.Reader(self.file_path)
-            self._logger.info(
-                "File loaded successfully from %s", self.file_path
-            )
-        except (InvalidDatabaseError, FileNotFoundError, ValueError):
+            new_db = maxminddb.open_database(self.file_path)
+        except (InvalidDatabaseError, OSError, ValueError):
             self._logger.exception(
                 "Failed to load maxmind db! Maybe the download was interrupted"
             )
+        else:
+            if self.db is not None:
+                self.db.close()
+
+            self.db = new_db
+            self._logger.info(
+                "File loaded successfully from %s", self.file_path
+            )
```

Comment:
We can move this to the try block after the `open_database` line, it's equivalent and perhaps more readable.

Reply:
Ah, nevermind, there's been talk about this earlier. I prefer early return in the 'except branch', but it's a nitpick, do what you think is best.

Reply:
Haha. Yea I really don't know what's better. Usually I put it inside the try-except, but Eoin suggested this way so I thought I'd give it a try.

Reply:
In my opinion this way we catch what we expect can fail, and anything else that starts failing with similar errors in the future will be a surprise to us instead of being quietly caught 🤷